flambe/nn/pooling.py

```diff
@@ -109,8 +108,16 @@ def forward(self,
 
         """
         # Apply pooling
-        if padding_mask is None:
-            padding_mask = torch.ones((data.size(0), data.size(1))).to(data)
+        padding_mask = padding_mask or _default_padding_mask(data)
+        value_count = padding_mask.sum(dim=1).unsqueeze(1) \
+            * torch.ones((data.size(0), data.size(2)))
```

Comment:
Why is `* torch.ones(...)` necessary?

Reply:
Because once the denominator (count of elements) is calculated, we need to divide EACH tensor value by that denominator, that's what the ones do

Reply:
Removed!